engine/common/system.c

```diff
@@ -376,11 +376,19 @@ void Sys_ShellExecute( const char *path, const char *parms, qboolean shouldExit
 					document.location.href = Pointer_stringify($0);
 				return 0;
 			}, (int)path );
-#elif (defined(__linux__) && !defined (__ANDROID__)) || defined (__FreeBSD__) || defined (__NetBSD__) || defined(__OpenBSD__) || defined(__APPLE__)
+#elif (defined(__linux__) && !defined (__ANDROID__)) || defined (__FreeBSD__) || defined (__NetBSD__) || defined(__OpenBSD__) || defined(__APPLE__) || defined(__HAIKU__)
 
 	if( !Q_strcmp( path, GENERIC_UPDATE_PAGE ) || !Q_strcmp( path, PLATFORM_UPDATE_PAGE ))
 		path = XASH_UPDATE_PAGE;
 
+	// Prevent "open: www.url.com: No such file or directory" error
+	char http[MAX_SYSPATH];
+	if( Q_strstr( path, "www" ) && !Q_strstr( path, "http" ) )
```

Comment:
Use Q_strncmp so you can check http at the beginning of string.

Reply:
Done!